File: GO/EvilAutocorrect.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

import "strings"

const YOURCLIENT = "your client"

var POINTS = []string{".", ",", "?", "!"}

/**
 * Transform all words 'you' to 'your client'.
 */
func autocorrect(input string) string {
	var result string = ""
	words := strings.Split(input, " ")

	for _, word := range words {
		if strings.ToLower(word) == "u" || strings.ToLower(word) == "you" {
			result += YOURCLIENT
		} else if len(word) == 3 && strings.ToLower(string(word[0:3])) == "you" && in_array(POINTS, string(word[len(word)-1])) {
			last := string(word[len(word)-1])

			result += YOURCLIENT + last
		} else if len(word) > 2 && strings.ToLower(string(word[0:3])) == "you" && checkU(string(word[3])) {
			result += YOURCLIENT

			last := string(word[len(word)-1])
			if in_array(POINTS, last) {
				result += last
			}
		} else {
			result += word
		}
		result += " "
	}

	return strings.TrimSpace(result)
}

func checkU(word string) bool {
	all := strings.Split(word, "")

	for i, letter := range all {
		if letter == "u" || len(all)-1 == i && in_array(POINTS, letter) {
			//return true;
		} else {
			return false
		}
	}

	return true
}

func in_array(data []string, el string) bool {
	for _, v := range data {
		if v == el {
			return true
		}
	}

	return false
}

File: PHP/EvilAutocorrect.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

//echo "The result is "+autocorrect( "We have sent the deliverables to you." );

class EvilAutocorrect {

    const YOURCLIENT = "your client";
    const POINTS = ['.',',','?','!'];

    /**
     * Transform all words 'you' to 'your client'.
     */
    function autocorrect( $input ) {
        $result = "";
        $words = explode(" ", $input);

        foreach ( $words as &$word ) {
            if ( "u" == strtolower( $word ) || "you" == strtolower( $word ) ) {
                $result .= self::YOURCLIENT;
            } else if ( strlen( $word ) == 3 && "you" == strtolower( substr($word, 0, 3) ) && in_array( substr($word,-1,1), self::POINTS ) ) {
                $last = substr($word,-1,1);

                $result .= self::YOURCLIENT . $last;
            } else if ( strlen( $word ) > 2 && "you" == strtolower( substr($word, 0, 3) ) && $this->checkU( substr($word, 3) ) ) {
                $result .= self::YOURCLIENT;

                $last = $word[ strlen( $word )-1 ];
                if ( in_array( $last, self::POINTS ) ) {
                    $result .= $last;
                }
            } else {
                $result .= $word;
            }
            $result .= " ";
        }

        return trim( $result );
    }

    function checkU( $word ) {
        $all = str_split( $word );
    
        for ( $i=0; $i < count( $all ); $i++ ) {
            $letter = $all[$i];
    
            if ( "u" == $letter || count( $all )-1 == $i && in_array( $letter, self::POINTS ) ) {
                //return true;
            } else {
                return false;
            }
        }
    
        return true;
    }

}
